src/core/worker.js

```diff
@@ -557,6 +557,19 @@ class WorkerMessageHandler {
       function ({ isPureXfa, numPages, annotationStorage, filename }) {
         pdfManager.requestLoadedStream();
 
+        const newAnnotationsByPage = new Map();
+        for (const [key, value] of annotationStorage) {
+          if (!key.startsWith("editor")) {
```

Comment:
This feels a little brittle, since there's nothing stopping a user from manually inserting something in the AnnotationStorage that begins with this string. However, I suppose that there's no other *simple* solution here given that structuredCloning is involved?

---

In any case, how about making sure that this begins with "editor-" instead?

Reply:
A possible solution could be to generate a random hash string and use it as a prefix for those specific annotation ids.
Or use something like "\\0e\\0\\d\\0i\\0t\\0o\\0r\\0-" (or "pdfjs::internal::editor-"), the problem is the same but likely it should minimize the risk of a conflict.
I thought using a Symbol but it doesn't work: 
https://html.spec.whatwg.org/multipage/structured-data.html#structuredserializeinternal

Reply:
> Or use something like "\0e\0\d\0i\0t\0o\0r\0-" (or "pdfjs::internal::editor-"), the problem is the same but likely it should minimize the risk of a conflict.

Either of them works for me, since as you say that would *reduce* the risk of accidental matches.
Also, you might want to add the string as a new constant in `src/shared/util.js` to avoid repeating it verbatim in multiple places?

> I thought using a Symbol but it doesn't work:
> https://html.spec.whatwg.org/multipage/structured-data.html#structuredserializeinternal

Yes, I wanted to suggest that but then remembered that it's not possible to structure clone Symbols :-(